server.go

```diff
@@ -243,6 +245,36 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
+// ServiceMetadata returns the metadata for a service or method.
+// service should be the full service name with package, in the form of <package>.<service>.
+// method should be the method name only.
+// If only service is important, method should be an empty string.
+func (s *Server) ServiceMetadata(service, method string) interface{} {
```

Comment:
Ah yes, that would make sense then. At the very least the function's documentation should likely say something about what the intended audience is. Off the top of my head, “function is used in other gRPC modules, and thus must be exported, it is however not intended for general use.”

This is just an example though! You certainly want to use better/clearer/considered language. ;)